code/modules/mob/living/silicon/pai/software.dm

```diff
@@ -268,29 +278,45 @@
 								M = E
 						src.securityActive1 = R
 						src.securityActive2 = M
-		if("securityhud")
-			if(href_list["toggle"])
-				src.secHUD = !src.secHUD
-		if("medicalhud")
-			if(href_list["toggle"])
-				src.medHUD = !src.medHUD
 		if("translator")
 			if(href_list["toggle"])
 				universal_speak = !universal_speak
 				universal_understand = !universal_understand
-		if("doorjack")
-			if(href_list["jack"])
-				if(src.cable && src.cable.machine)
-					src.hackdoor = src.cable.machine
-					src.hackloop()
+		if("wirejack")
 			if(href_list["cancel"])
-				src.hackdoor = null
-			if(href_list["cable"])
-				var/turf/T = get_turf(src.loc)
-				src.cable = new /obj/item/weapon/pai_cable(T)
-				for (var/mob/M in viewers(T))
-					M.show_message("<span class='warning'>A port on [src] opens to reveal [src.cable], which promptly falls to the floor.</span>", 3, "<span class='warning'>You hear the soft click of something light and hard falling to the ground.</span>", 2)
-	//src.updateUsrDialog()		We only need to account for the single mob this is intended for, and he will *always* be able to call this window
+				src.hacktarget = null
+		if("chemsynth")
+			if(href_list["chem"])
+				if(!istype(src.loc.loc,/mob/living/carbon))
+					src << "<span class='warning'>You must have a carrier to inject with chemicals!</span>"
+				else if(chargeloop("chemsynth"))
+					if(istype(src.loc.loc,/mob/living/carbon)) //Sanity
+						var/mob/living/M = src.loc.loc
+						M.reagents.add_reagent(href_list["chem"], 15)
+						playsound(get_turf(src.loc), 'sound/effects/bubbles.ogg', 50, 1)
+				else
+					src << "<span class='warning'>Charge interrupted.</span>"
+		if("foodsynth")
+			if(href_list["food"] && chargeloop("foodsynth"))
+				var/obj/item/weapon/reagent_containers/food/F
+				switch (href_list["food"])
+					if("donut")
+						F = new /obj/item/weapon/reagent_containers/food/snacks/donut/normal(get_turf(src))
+					if("banana")
+						F = new /obj/item/weapon/reagent_containers/food/snacks/grown/banana(get_turf(src))
+					else
+						F = new /obj/item/weapon/reagent_containers/food/snacks/badrecipe(get_turf(src))
+				if(istype(src.loc.loc,/mob))

```

Comment:
I made a proc for this once, you can use
var/mob/M = find_holder_of_type(src, /mob)
it returns 0 if it doesn't find anything so you can do if(M) M.put_in_hands(F)